### ISSUE TITLE
Confirming: No API Guarantee for Filter or Append

### ISSUE THREAD

**Issue by kemitchell** [maxogden/abstract-blob-store#29]
Would really appreciate confirmation that `createReadStream` and `createWriteStream` make no guarantees relevant for:

1. Appending data to a blob. To append a blob, users have to stream the existing data to a buffer, then stream it back to the store, with whatever's to be appended tacked on at the end.

2. Filtering data in a blob. This requires buffering, too.

Many thanks!

**Comment by jnordberg** [comment]
That's correct, the interface only guarantees that you'll get a readable or writable stream corresponding to a key. Implementations can use options to extend this though, for example by passing `{append: true}` to `createWriteStream`
